go/tools/generate_test_main.go

```diff
@@ -39,6 +59,29 @@ type Cases struct {
 	HasTestMain      bool
 	Version17        bool
 	Version18OrNewer bool
+	Cover            []coverInfo
```

Comment:
Could this be replaced with `CoverVars []CoverVar`? Seems like a flat list would work well here, since we only range over this. This would let you drop the `coverInfo` struct.